src/printer/relax_script_printer.cc

```diff
@@ -350,11 +350,17 @@ Doc RelaxScriptPrinter::VisitNode_(const relax::SeqExprNode* op) {
 Doc RelaxScriptPrinter::VisitNode_(const relax::FunctionNode* op) {
   Optional<String> gsymbol = op->GetAttr<String>(tvm::attr::kGlobalSymbol);
   if (gsymbol) {
-    ICHECK_EQ(gsymbol.value(), relax_func_name_);
-    return PrintFunctionDef(Doc::Text(relax_func_name_), GetRef<relax::Function>(op),
+    if (!relax_func_name_) {
+        relax_func_name_ = gsymbol.value();
+    }
+    else {
+        ICHECK_EQ(gsymbol.value(), relax_func_name_.value());
+    }
+    return PrintFunctionDef(Doc::Text(relax_func_name_.value()), GetRef<relax::Function>(op),
                             /*is_global=*/true);
   } else {
-    return PrintFunctionDef(Doc::Text(relax_func_name_), GetRef<relax::Function>(op),
+    String print_name = relax_func_name_ ? relax_func_name_.value() : "foo";
```

Comment:
And we have a `value_or` method.
```suggestion
    String print_name = relax_func_name_.value_or(relax_default_func_name_);
```